packages/server/src/api/controllers/deploy/index.js

```diff
@@ -94,18 +94,16 @@ async function initDeployedApp(prodAppId) {
 }
 
 async function deployApp(deployment) {
-  try {
-    const appId = getAppId()
-    const devAppId = getDevelopmentAppID(appId)
-    const productionAppId = getProdAppID(appId)
-
-    const replication = new Replication({
-      source: devAppId,
-      target: productionAppId,
-    })
+  const appId = getAppId()
+  const devAppId = getDevelopmentAppID(appId)
+  const productionAppId = getProdAppID(appId)
+  const replication = new Replication({
```

Comment:
could use `let replication` outside the try and do the rest inside the try like we had before so we can catch any of exceptions there, then in the finally wrap the close with `if (replication)`